pkg/netpol/eval/check.go

```diff
@@ -115,10 +103,9 @@ func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPe
 		if err != nil {
 			return nil, err
 		}
-		// if src and dst are the same workload peer, their conversion to pods should be of different pods
-		// (if owner has more than 1 instances)
+		// if src and dst are the same workload peer, it is a self-loop connection we want to skip it
 		if srcPeer.String() == dstPeer.String() {
-			pe.changePodPeerToAnotherPodObject(dstPodPeer)
+			return nil, nil
```

Comment:
I prefer to maintain here the previous returned value (based on `changePodPeerToAnotherPodObject`) , so that package `eval` does return the most accurate connectivity result, and only package `connlist` chooses to avoid evaluating allowed connections between two same workloads.